### ISSUE TITLE
Recommend contains() for find().is_some()

### ISSUE THREAD

**Issue by pickfire** [rust-lang/rust-clippy#6010]
### What it does

Recommend `.contains()` if `find().is_some()` is used on a `&str` (including `String`) and `slice`.

### Categories (optional)

- Kind: complexity

*What is the advantage of the recommended code over the original code*

For example:
- Improve readibility

### Drawbacks

None.

### Example

```rust
"hello world".find("world").is_some();
```

Could be written as:

```rust
"hello world".contains("world");
```

**Comment by longlb** [comment]
I'd like to tackle this one.

**Comment by longlb** [comment]
@pickfire How would you call `find().is_some()` against a slice? Would it be something like `slice.iter().find().is_some()` or another way, because there is a currently a lint for the iter case that recommends `any()` be used in its place.

**Comment by pickfire** [comment]
What you said is correct but later when array comes out to have find we need to add it too.

**Comment by rsulli55** [comment]
I am interested on working on this, if @longlb has moved on.   I have a good start and should be able to submit a pull request tomorrow.

**Comment by longlb** [comment]
Go for it, I'd actually love to see how you did it because I've been unable to figure out a lot of the compiler stuff.

**Comment by rsulli55** [comment]
Ok, great!  

@pickfire So I made a new lint to do this (basically following the tutorial), but I am wondering if it would be better to instead add this as an extra case of `lint_search_is_some()` in `methods/mod.rs`

**Comment by pickfire** [comment]
@rsulli55 Not quite sure but probably good. Maybe you can take inspiration from `find_map()` https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/methods/mod.rs#L2905?

**Comment by rsulli55** [comment]
@pickfire Ok, thanks!  I opened a WIP PR for now with what I have but I will be happy to move the code around if it should be placed somewhere else.
